ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/reloperators/HiveTableScan.java

```diff
@@ -263,6 +279,10 @@ public boolean isInsideView() {
     return insideView;
   }
 
+  public boolean isFetchDeletedRows() {
+    return fetchDeletedRows;
+  }
+
   // We need to include isInsideView inside digest to differentiate direct
   // tables and tables inside view. Otherwise, Calcite will treat them as the same.
   // Also include partition list key to trigger cost evaluation even if an
```

Comment:
The `computeDigest` method should include `fetchDeletedRows` in the computed digest.
Otherwise, if you would have a plan with two TS on the same table, one with `fetchDeletedRows` set to `true` and another one set to `false`, the planner may merge those TS operators and take a single one of them.
Maybe you can try to reproduce it with a custom test and add it to the patch.